tests/TestCase/ORM/TableTest.php

```diff
@@ -6702,4 +6704,49 @@ public function skipIfSqlServer()
             'SQLServer does not support the requirements of this test.'
         );
     }
+
+    public function testJoinOrdersWithAlias()
+    {
+        $expected = [
+            0 => [
+                'id' => 1,
+                'category' => 1,
+                'name' => 'First product',
+                'price' => 10,
+                'Order' => [
+                    'id' => '1',
+                    'product_category' => '1',
+                    'product_id' => '1',
+                ],
+            ],
+        ];
+        $table = $this->getTableLocator()->get('products');
+        $options = [
+            'fields' => [
+                'products.id',
+                'products.category',
+                'products.name',
+                'products.price',
+                'Order.id',
+                'Order.product_category',
+                'Order.product_id',
+            ],
+            'join' => [
+                [
+                    'table' => 'orders',
+                    'alias' => 'Order',
+                    // With escaped alias the test passes.
+                    //'alias' => '`Order`',
+                    'type' => 'INNER',
+                    'conditions' => [
+                        'Order.product_id = products.id',
```

Comment:
This string condition won't be identifier quoted by the ORM.

Reply:
This condition creates no problems. The test passes with that. But the alias is a problem.

Reply:
This test will never fully pass right now, as it will be run with and without identifier quoting. Without identifier quoting this query is always invalid.

With identifier quoting enabled the generated query is 

```sql
'SELECT "products"."id" AS "products__id", "products"."category" AS "products__category", "products"."name" AS "products__name", "products"."price" AS "products__price", "Order"."id" AS "Order__id", "Order"."product_category" AS "Order__product_category", "Order"."product_id" AS "Order__product_id" FROM "products" "products" INNER JOIN "orders" "Order" ON Order.product_id = products.id'
```

The only unquoted Order alias is the one in your condition string.

Reply:
Not really.

With `'alias' => 'Order',` this test fails:
```
/src/Database/Statement/MysqlStatement.php (line 37)
########## DEBUG ##########
object(PDOStatement) {
	queryString => 'SELECT products.id AS `products__id`, products.category AS `products__category`, products.name AS `products__name`, products.price AS `products__price`, Order.id AS `Order__id`, Order.product_category AS `Order__product_category`, Order.product_id AS `Order__product_id` FROM products products INNER JOIN orders Order ON Order.product_id = products.id'
}
```

With `'alias' => '`Order`',` this test passes:
```
/src/Database/Statement/MysqlStatement.php (line 37)
########## DEBUG ##########
object(PDOStatement) {
	queryString => 'SELECT products.id AS `products__id`, products.category AS `products__category`, products.name AS `products__name`, products.price AS `products__price`, Order.id AS `Order__id`, Order.product_category AS `Order__product_category`, Order.product_id AS `Order__product_id` FROM products products INNER JOIN orders `Order` ON Order.product_id = products.id'
}
```

You can copy-paste these queries to MySQL Workbench to see that the first one is invalid but the second one is valid.

Reply:
The failing query is generated with quoting disabled. The test suite runs all db related tests twice, once with auto quoting disabled and once with enabled. 

Add this statement `$table->getConnection()->getDriver()->enableAutoQuoting(true);` before doing the find call and you won't get any failure.

We already have tests checking that both the table name and alias are quoted with auto quoting enabled.

https://github.com/cakephp/cakephp/blob/448f30918f4a16fc288d84b54bba885d60fe3ebc/tests/TestCase/Database/QueryTest.php#L4059-L4065